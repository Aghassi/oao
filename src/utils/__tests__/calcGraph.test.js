/* eslint-env jest */

import calcGraph from '../calcGraph';

const ALL_SPECS_NO_CYCLE = {
  a: {
    name: 'a',
    specs: {
      dependencies: { b: '*', c: '*' },
    },
  },
  b: {
    name: 'b',
    specs: {
      dependencies: { ext: '*' },
    },
  },
  c: {
    name: 'c',
    specs: {
      dependencies: { ext: '*' },
      devDependencies: { d: '*' },
    },
  },
  d: {
    name: 'd',
    specs: {},
  },
};

const ALL_SPECS_CYCLE = {
  a: {
    name: 'a',
    specs: {
      dependencies: { b: '*', c: '*' },
    },
  },
  b: {
    name: 'b',
    specs: {
      dependencies: { ext: '*' },
    },
  },
  c: {
    name: 'c',
    specs: {
      dependencies: { ext: '*' },
      devDependencies: { d: '*', e: '*' },
    },
  },
  d: {
    name: 'd',
    specs: {},
  },
  e: {
    name: 'e',
    specs: {
      optionalDependencies: { a: '*' },
    },
  },
};

describe('buildGraph', () => {
  it('calculates a directed acyclic graph correctly when no cycles are present', () => {
    const dag = calcGraph(ALL_SPECS_NO_CYCLE);
    expect(dag).toMatchSnapshot();
  });

  it('handles cycles correctly', () => {
    const dag = calcGraph(ALL_SPECS_CYCLE);
    expect(dag).toMatchSnapshot();
  });
});
