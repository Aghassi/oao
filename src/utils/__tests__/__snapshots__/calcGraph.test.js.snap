// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildGraph calculates a directed acyclic graph correctly when no cycles are present 1`] = `
Array [
  "b",
  "d",
  "c",
  "a",
]
`;

exports[`buildGraph handles cycles correctly 1`] = `
Array [
  "b",
  "d",
  "e",
  "c",
  "a",
]
`;
